src/systemcmds/mtd/mtd.cpp

```diff
@@ -380,18 +389,43 @@ mtd_start(mtd_instance_s &instance, const char *partition_names[], unsigned n_pa
 	unsigned offset;
 	unsigned i;
 
-	for (offset = 0, i = 0; i < n_partitions; offset += nblocks, i++) {
+
+	instance.partition_block_counts = new int[n_partitions];
+
+	if (instance.partition_block_counts == nullptr) {
+		PX4_ERR("mtd_partition failed allocation counts");
+		return 1;
+	}
+
+	instance.partition_names = new const char *[n_partitions];
+
+	if (instance.partition_names == nullptr) {
+		PX4_ERR("mtd_partition failed allocation for names");
+		return 1;
+	}
+
+	for (unsigned int n = 0; n < n_partitions; n++) {
+		float percentage = instance.partition_percentages == nullptr  ?
+				   100.0f / n_partitions : instance.partition_percentages[n];
+		volatile int nb = neraseblocks * (percentage / 100.0f);
+		instance.partition_block_counts[n] = (nb == 0 ? 1 : nb);
```

Comment:
instance.partition_block_counts[n] = (nb == 0 ? 1 : nb) * blkpererase;

Reply:
Going forward the % have been removed from the system. The partition layout can be fully specified or the default RAMTRON configurations can be used. See:   https://github.com/PX4/PX4-Autopilot/pull/16212. 

So this code is not used in any config and all devices PX4 is supporting are  blkpererase = 1.

Reply:
Yes, we had moved the discussion to #16212.